torch/utils/tensorboard/writer.py

```diff
@@ -0,0 +1,731 @@
+            scalar_value = workspace.FetchBlob(scalar_value)
+        self.get_file_writer().add_summary(
+            scalar(tag, scalar_value), global_step, walltime)
+
+    def add_scalars(self, main_tag, tag_scalar_dict, global_step=None, walltime=None):
+        """Adds many scalar data to summary.
+
+        Note that this function also keeps logged scalars in memory. In extreme case it explodes your RAM.
+
+        Args:
+            main_tag (string): The parent name for the tags
+            tag_scalar_dict (dict): Key-value pair storing the tag and corresponding values
+            global_step (int): Global step value to record
+            walltime (float): Optional override default walltime (time.time()) of event
+
+        Examples::
+
+            writer.add_scalars('run_14h', {'xsinx':i*np.sin(i/r),
+                                           'xcosx':i*np.cos(i/r),
+                                           'arctanx': numsteps*np.arctan(i/r)}, i)
+            # This call adds three values to the same scalar plot with the tag
+            # 'run_14h' in TensorBoard's scalar section.
+        """
+        walltime = time.time() if walltime is None else walltime
+        fw_logdir = self.get_file_writer().get_logdir()
+        for tag, scalar_value in tag_scalar_dict.items():
+            fw_tag = fw_logdir + "/" + main_tag + "/" + tag
+            if fw_tag in self.all_writers.keys():
+                fw = self.all_writers[fw_tag]
+            else:
+                fw = FileWriter(logdir=fw_tag)
+                self.all_writers[fw_tag] = fw
+            if self._check_caffe2_blob(scalar_value):
+                scalar_value = workspace.FetchBlob(scalar_value)
+            fw.add_summary(scalar(main_tag, scalar_value),
+                           global_step, walltime)
+            self.__append_to_scalar_dict(
+                fw_tag, scalar_value, global_step, walltime)
+
+    def export_scalars_to_json(self, path):
+        """Exports to the given path an ASCII file containing all the scalars written
+        so far by this instance, with the following format:
+        {writer_id : [[timestamp, step, value], ...], ...}
+
+        The scalars saved by ``add_scalars()`` will be flushed after export.
+        """
+        with open(path, "w") as f:
+            json.dump(self.scalar_dict, f)
+        self.scalar_dict = {}
+
+    def add_histogram(self, tag, values, global_step=None, bins='tensorflow', walltime=None, max_bins=None):
+        """Add histogram to summary.
+
+        Args:
+            tag (string): Data identifier
+            values (torch.Tensor, numpy.array, or string/blobname): Values to build histogram
+            global_step (int): Global step value to record
+            bins (string): one of {'tensorflow','auto', 'fd', ...}, this determines how the bins are made. You can find
+              other options in: https://docs.scipy.org/doc/numpy/reference/generated/numpy.histogram.html
```

Comment:
I think you actually wanted to link to https://docs.scipy.org/doc/numpy/reference/generated/numpy.histogram_bin_edges.html#numpy.histogram_bin_edges no?

Reply:
This also accepts non-string input too, no?

Reply:
1. Yes, that link is more specific.
2. "This also accepts non-string input too?": yes. Replied above :)